src/index.ts

```diff
@@ -158,6 +159,26 @@ class Storyblok {
 		})
 	}
 
+	public customFetch(
+		slug: string,
+		params?: ISbStoriesParams,
+		fetchOptions?: object
+	): Promise<any> {
+		if (fetchOptions && Object.keys(fetchOptions).length > 0) {
+			this.fetchOptions = fetchOptions
+		}
+
+		if (!params) params = {} as ISbStoriesParams
+
+		const url = `/${slug}`
+		const query = this.factoryParamOptions(url, params)
+
+		const response = this.cacheResponse(url, query)
```

Comment:
@thiagosaife as `cacheResponse` is an async method, we should `await` and clean the `this.fetchOptions` object in the `finally` of the `try/catch` - otherwise it can lead to a race condition.

May I ask.. why not just passing the `fetchOptions` as a param of the call? As they are unique per call, it'll make it simpler and cleaner. If you store it in the instance you must take care of cleaning and might be more error-prone.

Reply:
Makes sense, thanks for elaborating on the details @thiagosaife. For me, if we have good tests backing up this functionality, we're good to go 🚀